src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_config.h"
+
+#if HAVE_LINUX_RTNETLINK_H
+#include "pal_types.h"
+#include "pal_networkchange.h"
+#include "pal_utilities.h"
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/uio.h>
+#include <sys/socket.h>
+#include <net/if.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+
+#pragma clang diagnostic ignored "-Wcast-align"
+
+extern "C" int CreateNetworkChangeListenerSocket()
```

Comment:
It seems that the return value of socket is fairly limited, so putting it into an int32 seems to be safe enough.